Tests/AsyncHTTPClientTests/HTTPClientTestUtils.swift

```diff
@@ -484,7 +485,11 @@ internal final class HTTPBin<RequestHandler: ChannelInboundHandler> where
                     // Successful upgrade to HTTP/2. Let the user configure the pipeline.
                     let http2Handler = NIOHTTP2Handler(
                         mode: .server,
-                        initialSettings: NIOHTTP2.nioDefaultSettings
+                        initialSettings: [
+                            // TODO: make max concurrent streams configurable
+                            HTTP2Setting(parameter: .maxConcurrentStreams, value: 10),
```

Comment:
`maxConcurrentStreams` will be made configurable and dynamically changeable in a follow up PR.